engine/src/test/java/io/camunda/zeebe/engine/processing/message/PublishMessageTest.java

```diff
@@ -190,9 +190,9 @@ public void shouldExpireMessageAfterTTL() {
             .getFirst();
 
     Assertions.assertThat(deletedEvent.getValue())
-        .hasName("order canceled")
-        .hasCorrelationKey("order-123")
-        .hasTimeToLive(100L)
+        .hasName("")
+        .hasCorrelationKey("")
+        .hasTimeToLive(-1L)
         .hasMessageId("");
```

Comment:
❌ This test doesn't verify that we delete the correct message, as we don't assert the key. We should add an assertion:

`Assertions.assertThat(deletedEvent).hasKey(publishedRecord.getKey());`

Reply:
@remcowesterhoud, thanks for raising it. I adjusted the test case accordingly 69ef1e9da1cbca03fcbd70419699cac793f24610

> I also noticed that if the TTL is <= 0 we [still write the full message](https://github.com/camunda/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessagePublishProcessor.java#L123-L126).